canonicalwebteam/http/__init__.py

```diff
@@ -100,24 +102,62 @@ class UncachedSession(BaseSession, requests.Session):
     pass
 
 
-class CachedSession(BaseSession, requests_cache.CachedSession):
+class CacheAdapterWithTimeout(CacheControlAdapter):
     """
-    A session object for making HTTP requests with cached responses.
+    A combination of the CacheControl and Timeout adapter
+    to provide both functionalities.
+    """
+
+    def __init__(self, heuristic, cache, timeout=None, *args, **kwargs):
+        self.timeout = timeout
+        self.cache_control_adapter = CacheControlAdapter(
```

Comment:
Here you're creating a class that is extended from `CacheControlAdapter`, but then actually creating the controller as a property of the same class. That's weird.

What you should be doing here is using the parent (`CacheControlAdapter`)'s constructor to set up your extended object properly:

``` python3
class CacheAdapterWithTimeout(CacheControlAdapter):
    """
    A combination of the CacheControl and Timeout adapter
    to provide both functionalities.
    """

    def __init__(self, cache, heuristic, timeout=None, *args, **kwargs):
        self.timeout = timeout
        return super(CacheAdapterWithTimeout, self).__init__(
            cache=cache, heuristic=heuristic, *args, **kwargs
        )

    def send(self, *args, **kwargs):
        kwargs['timeout'] = self.timeout
        return super(CacheAdapterWithTimeout, self).send(*args, **kwargs)
```

(The `super(CacheAdapterWithTimeout, self).__init__()` syntax is ugly. It only needs to be so verbose to support Python 2 - once we can drop Python 2 suppoer, this can become simply `super().__init__()`, which is nicer)